### PR TITLE
Allow access to APIs only from local and valid networks

### PR DESCRIPTION
**Pull request by Stell0**

### CODE REVIEW

**Review by DavidePrincipi — COMMENTED** [review]
Do we want to restrict access to `/provisioning` as well?

**Comment by Stell0** [comment]
I'm not sure about that, maybe it's better:
- /provisining/MAC could leak phone sip credential and it's easy to bruteforce MAC
- /provisioning/token/FILE should be considered secure
could we restrict only access to /provisining/MAC? If it's possible I think it's the best option, otherwise I'd restrict.
It's always possible to allow access from NethServer GUI, but it's intended for administrator access to interface, not users

**Comment by Stell0** [comment]
@DavidePrincipi what about that?

File: root/etc/e-smith/templates/etc/httpd/conf.d/tancredi.conf/10base [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    our $localAccess = join(' ',@localAccessSpec);
+
+    our $validFrom = ${'httpd-fpbx'}{'ValidFrom'};
+    $validFrom =~ s/,/ /g;
+
+    $OUT = "";
+}
+
+<LocationMatch "/provisioning/[a-zA-Z0-9\.\-_]+$">
+    Require ip {$localAccess} {$validFrom}
```
Comment (DavidePrincipi) [inline root/etc/e-smith/templates/etc/httpd/conf.d/tancredi.conf/10base:15]: Does Apache need a ProxyPass directive here? I'm afraid the next Location stanza is not evaluated if the regexp matches...

**Review by DavidePrincipi — COMMENTED** [review]
Apache config doubt...

File: root/etc/e-smith/templates/etc/httpd/conf.d/tancredi.conf/10base (hunk 1 above)
Comment (Stell0) [reply root/etc/e-smith/templates/etc/httpd/conf.d/tancredi.conf/10base:15]: This configuration works:
```
<LocationMatch "/provisioning/[a-zA-Z0-9\.\-_]+$">
    Require ip 127.0.0.1 192.168.122.75
</LocationMatch>

<Location "/provisioning">
    ProxyPass "fcgi://127.0.0.1:9605/usr/share/tancredi/public/provisioning.php"
</Location>
```

This doesn't:
```
<LocationMatch "/provisioning/[a-zA-Z0-9\.\-_]+$">
    Require ip 127.0.0.1 192.168.122.75
    ProxyPass "fcgi://127.0.0.1:9605/usr/share/tancredi/public/provisioning.php"
</LocationMatch>

<Location "/provisioning">
    ProxyPass "fcgi://127.0.0.1:9605/usr/share/tancredi/public/provisioning.php"
</Location>
```
It fails with error `ProxyPass/<Proxy> and ProxyPassMatch/<ProxyMatch> can't be used altogether with the same worker name (fcgi://127.0.0.1:9605/usr/share/tancredi/public/provisioning.php)` 
but httpd documentation isn't really clear to me https://httpd.apache.org/docs/current/mod/core.html#locationmatch

**Review by DavidePrincipi — APPROVED** [review]

**Pull request merged**